### ISSUE TITLE
Raise warnings on use, not import

### ISSUE THREAD

**Issue by ljwolf** [pysal/pysal#1139]
Right now we raise warnings for optional dependencies at import in `segregation` and `spvcm` when imported. 

We should raise a UserError when these are used, not when the package is imported.

**Comment by renanxcortes** [comment]
Yes, I agree that this might be a good strategy. We are thinking about doing this in `tobler` where in some very specific options we might want to use `xgboost` and `shap`. Would you have some example of how we can address that in the code? Would this be inside the class/function that uses `xgboost` something like:

```
try:
    import xgboost
except:
    UserError('This is the error message.')
```

?

**Comment by ljwolf** [comment]
Below the function, yes. At the module level, no. So, something like:

```python
def do_something_cool():
    try:
        import xgboost
    except ImportError as e:
        raise ImportError('xgboost is required to do something cool.')
    ...
```

**Comment by knaaptime** [comment]
this is resolved as of version 2.2
